app/src/pages/datasets/DatasetActionMenu.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  ReactNode,
+  startTransition,
+  useCallback,
+  useState,
+} from "react";
+import { graphql, useMutation } from "react-relay";
+
+import {
+  ActionMenu,
+  Button,
+  Dialog,
+  DialogContainer,
+  Flex,
+  Icon,
+  Icons,
+  Item,
+  Text,
+  View,
+} from "@arizeai/components";
+
+import { DatasetActionMenuDeleteMutation } from "./__generated__/DatasetActionMenuDeleteMutation.graphql";
+
+type DatasetActionMenuProps = {
+  datasetId: string;
+  datasetName: string;
+  onDatasetDelete: () => void;
+  onDatasetDeleteError: (error: Error) => void;
+};
+
+enum DatasetAction {
+  DELETE = "deleteDataset",
+}
+
+export function DatasetActionMenu(props: DatasetActionMenuProps) {
+  const { datasetId, datasetName, onDatasetDelete, onDatasetDeleteError } =
+    props;
+  const [confirmDialog, setConfirmDialog] = useState<ReactNode>(null);
+  const [commitDelete, isCommittingDelete] =
+    useMutation<DatasetActionMenuDeleteMutation>(graphql`
+      mutation DatasetActionMenuDeleteMutation($datasetId: GlobalID!) {
+        deleteDataset(input: { datasetId: $datasetId }) {
+          __typename
+        }
+      }
+    `);
+  const handleDelete = useCallback(() => {
+    startTransition(() => {
+      commitDelete({
+        variables: {
+          datasetId,
+        },
+        onCompleted: () => {
+          onDatasetDelete();
+        },
+        onError: (error) => {
+          onDatasetDeleteError(error);
+        },
+        updater: (store) => {
+          // Invalidate the dataset in the store
+          const datasetNode = store.get(datasetId);
```

Comment:
For some reason I can't get the refetch to work deterministically. This deletes the node from the relay store which seems to do the trick...